### PR TITLE
Feat: Attach NFT media from connected wallet to Media Gallery

### LINKED ISSUE #449: Add a way for users to easily attach NFT media from their connected wallet to their Media Gallery
**Overview**
We want to make it easier for parcel licensors to anchor content that they care about to the Geo Web. We can integrate an NFT metadata querying service to return the NFTs owned by the currently connected wallet for easy pinning & anchoring to a parcel's Media Gallery.

**Requirements**
- Add a secondary "upload" button on the Media Gallery screen (https://github.com/Geo-Web-Project/cadastre/tree/main/components/gallery)
  - Icon = https://fonts.google.com/icons?selected=Material+Symbols+Outlined:account_balance_wallet:FILL@0;wght@400;GRAD@0;opsz@48&icon.query=wallet
  - Tooltip Text = "Select from Wallet"
  - Add a tooltip to the existing upload button = "Device Upload" 
- When the "Select from Wallet" button is clicked show a new "NFTs Held by..." view (see mockups below)
  - Utilize the https://www.infura.io/platform/nft-api to return the media and Name/ID of the user's wallet (minimally ChainIDs = 1 & 10)
- When the user confirms the selection of one of their NFTs on the new view, populate the CID & Name on the Media Gallery entry screen 

Media Gallery - Mobile
Media Gallery - Desktop

**Instructions**
Please base your PR off the latest `develop` branch and request review from both @tnrdd & @gravenp.

### PR DESCRIPTION
**Pull request by jordang7**
# Description

Secondary upload button on media gallery screen added, opens up a modal when pressed that displays all owned NFTs of that connected wallet through Alchemy NFT API. 
When the user selects an NFT, it takes the NFT image file and uploads to web3storage and populates the CID and the name.

# Issue

fixes #449 

dework task - https://app.dework.xyz/geo-web/cadastre?taskId=07cf1133-493e-4ef4-b986-14ecf1bba69e

# Checklist:

- [x] My commit message follows the Conventional Commits specification
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have tested my code
- [x] My changes generate no new warnings
- [x] My PR is rebased off the most recent `develop` branch
- [x] My PR is opened against the `develop` branch

# Alert Reviewers

@codynhat @gravenp @tnrdd

### CODE REVIEW

**Review by gravenp — CHANGES_REQUESTED** [review]
Nice work, @jordang7. The core functionality is working as expected. Just a couple of items to take a look at:

- The "Wallet Icon" on the new button on the Media Gallery page isn't loading for me
- Could we add support to pull NFTs from Ethereum mainnet in addition to the network that the user is currently connected to? I think most users will have the bulk of their collections there.
- There's a noticeable delay between clicking `Select` on the "NFTs Held by..." page and returning to the "Media Gallery" page. Seems like there's some processing going on before navigation. Since there is already a spinner on the "Media Gallery" page, is it possible to navigate instantly and have all the processing take place with that pending spinner shown? If that's not appropriate, we should add a spinner to the `Select` button to let the user know something is going on. I found myself clicking multiple times thinking it didn't work.
-  This might be specific to the test NFT and can be ignored but when I `Select` one of the [MultiFaucet Test NFTs](https://goerli-optimism.etherscan.io/token/0xf5de760f2e916647fd766b4ad9e85ff943ce3a2b?a=0xed6a062fbe2993be323af118f79e9b213c81f4f2) in my wallet, the following error is thrown in the console (If this error is "valid", then let's show a warning message for why the NFT can't be used):

9608-7b42b2dd4bded059.js:1 Uncaught (in promise) Error: Request failed with status code 400
    at e.exports (9608-7b42b2dd4bded059.js:1:660825)
    at e.exports (9608-7b42b2dd4bded059.js:1:663274)
    at XMLHttpRequest.b (9608-7b42b2dd4bded059.js:1:656076)

**Comment by jordang7** [comment]
Thanks for the feedback @gravenp. 
> * The "Wallet Icon" on the new button on the Media Gallery page isn't loading for me
- Looks like I forgot to commit the icon file. I will add it.
> * Could we add support to pull NFTs from Ethereum mainnet in addition to the network that the user is currently connected to? I think most users will have the bulk of their collections there.
- Makes sense, I will look into a way to do this with the Alchemy sdk.
> * There's a noticeable delay between clicking `Select` on the "NFTs Held by..." page and returning to the "Media Gallery" page. Seems like there's some processing going on before navigation. Since there is already a spinner on the "Media Gallery" page, is it possible to navigate instantly and have all the processing take place with that pending spinner shown? If that's not appropriate, we should add a spinner to the `Select` button to let the user know something is going on. I found myself clicking multiple times thinking it didn't work.
- Yes I agree with this. We can make it instantly navigate back to the main media gallery modal for a better user experience. 

I will try to get these changes done by either tomorrow night or Monday.

**Review by gravenp — COMMENTED** [review]
Thanks @jordang7. The changes you pushed addressed my first 3 bullets, so we're looking good there. 

I was still seeing 400 & "Network Errors" at a high rate for specific collections (although the messaging was better), so I chatted with Cody & Stefano during office hours about it. The root of the issue is that some NFTs utilize non-IPFS methods for linking their media. @codynhat is going to drop in some suggestions into a review for a way to simplify linking and handle media that's not IPFS-based.

**Comment by codynhat** [comment]
Hey @jordang7. Thanks for the PR! We met on a call earlier today to discuss this. The main outstanding issue is that this currently assumes that NFT metadata are all IPFS URIs and attempts to re-upload assets to Web3.storage. To make this cleaner and easier, we think it is best to just take the URI from NFT metadata and insert directly into the `content` field of the `MediaObject`. There is no need to re-upload or check the URI at all. Browsers will read the URI and decide how to fetch the content.

File: components/gallery/GalleryForm.tsx [hunk 1]
```diff
@@ -132,6 +152,49 @@ function GalleryForm(props: GalleryFormProps) {
...
+
+      const file = new File([blob], `${selectedNFT.title}.${fileExt}`, {
+        type: type,
+      });
+
+      const format = getFormat(file.name);
+      const { encoding } = format;
+      setFileFormat(encoding ?? null);
+      // Upload to Web3 storage
+      const added = await uploadFile(w3InvocationConfig, file);
```
Comment (codynhat) [inline components/gallery/GalleryForm.tsx:181]: As mentioned, this is no longer needed. The `selectedNFT?.media?.[0]?.raw` value can be passed directly to `content` below.

File: components/gallery/GalleryForm.tsx [hunk 2]
```diff
@@ -186,32 +249,93 @@ function GalleryForm(props: GalleryFormProps) {
...
+            justifyContent: "center",
+            alignItems: "center",
+          }}
+        />
+      )}
+
       <Form id="galleryForm" className="pt-2 text-start">
         <Row className="px-1 px-sm-3 d-flex align-items-end">
           <Col sm="12" lg="6" className="mb-3">
             <Form.Text className="text-primary">CID</Form.Text>
```
Comment (codynhat) [inline components/gallery/GalleryForm.tsx:273]: In addition, could you change the text field label to "URI"?

**Review by codynhat — CHANGES_REQUESTED** [review]

File: components/gallery/GalleryForm.tsx (hunk 1 above)
Comment (jordang7) [reply components/gallery/GalleryForm.tsx:181]: Hey Cody, thanks for the guidance. I am testing this method now and am getting a CID parsing error when I click the Done button to add the asset. 
image

Here is the media object of the test NFT I am using. How do you recommend dealing with a case like this that has the file path included in the cid? Should I remove both the ipfs:// and the filename if present and pass that directly to content?

{
    "gateway": "https://nft-cdn.alchemy.com/opt-goerli/36bba54ec895793b9993062ca2ca6a7e",
    "thumbnail": "https://res.cloudinary.com/alchemyapi/image/upload/thumbnailv2/opt-goerli/36bba54ec895793b9993062ca2ca6a7e",
    "raw": "ipfs://QmVdntsdbBPjrXoHuLSxk2Jr6HwZZD7MoJRguRXDeGU4sb/punks.webp",
    "format": "webp",
    "bytes": 266
}

Comment (codynhat) [reply components/gallery/GalleryForm.tsx:181]: Looks like the `newMediaGallery` prop is using the `MediaObject` type which is expecting a `CID` type for the `content` field. You could try just removing/replacing this type with one that expects a string instead, and remove the `CID.parse`.

Comment (codynhat) [reply components/gallery/GalleryForm.tsx:181]: The filename should be fine and can be kept. As long as the string is a URI that begins with a protocol scheme.

**Comment by codynhat** [comment]
Hey @jordang7. Looks like I messed something up with the build settings for `geo-web-cadastre-testnet`. We are going to review this preview (https://bafybeieixxalspz732cp5a55lx4rnrdyfspp6m4suui537jd6a3ml2zuoe.on.fleek.co/) at `geo-web-cadastre-staging`, which uses mainnet config.

**Pull request merged**